### DISCUSSION TITLE
Am I supposed to always use python-lsp-server from a virtualenv?

### DISCUSSION THREAD

**Discussion by lypwig** [python-lsp/python-lsp-server#603] [Q&A]
Hello,

I recently gave a try to [lapce](https://lap.dev/lapce/) among with python-lsp-server.

I first installed the lsp on my system, but I noticed that while I can go to definitions of my own project, I can not go to definitions of the project dependencies.

Then I tried to install the lsp on my venv (poetry) and updated the lsp path, and this time I was able to go to dependency definitions.

I don't really know how a lsp work, but am I supposed to alway install the lsp of the project venv? Or is it a problem/misconfiguration with my ide?

Thank you.

**Comment by smheidrich** [comment]
I was asking myself the same thing and after looking through similar discussions wrote a PR to add docs that address this, maybe it'll help:

- https://github.com/python-lsp/python-lsp-server/pull/643
